src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory_Methods.vb

```diff
@@ -276,6 +276,50 @@ Namespace Microsoft.CodeAnalysis.Semantics
+            If eventInvocation Is Nothing OrElse eventInvocation.ReceiverOpt Is Nothing
+                Debug.Assert(raiseEventStatement.HasErrors)
+                Return Create(raiseEventStatement.EventInvocation)
+            End If 
+
+            Dim receiver = eventInvocation.ReceiverOpt
+            Dim syntax = receiver.Syntax
+            Dim constantValue As [Optional](Of Object) = ConvertToOptional(receiver.ConstantValueOpt)
+            Dim boundInstance As BoundNode
+            If receiver.Kind = BoundKind.FieldAccess
+                Dim eventFieldAccess = CType(eventInvocation.ReceiverOpt, BoundFieldAccess)
+                Debug.Assert(eventFieldAccess.FieldSymbol.AssociatedSymbol = eventSymbol)
+
+                boundInstance = eventFieldAccess.ReceiverOpt
+            Else 
+                boundInstance = receiver
+            End If
+
+            Dim instance As Lazy(Of IOperation) = New Lazy(Of IOperation)(Function() If(eventSymbol.IsShared, Nothing, Create(boundInstance)))
+            
+            Dim EventReference As Lazy(Of IEventReferenceExpression) = New Lazy(Of IEventReferenceExpression)(Function() As IEventReferenceExpression
+                                                                                                                  Return New LazyEventReferenceExpression(eventSymbol, 
+                                                                                                                                                          instance, 
+                                                                                                                                                          eventSymbol, 
+                                                                                                                                                          _semanticModel, 
+                                                                                                                                                          syntax, 
+                                                                                                                                                          eventSymbol.Type, 
+                                                                                                                                                          constantValue, 
+                                                                                                                                                          eventInvocation.WasCompilerGenerated)
+                                                                                                              End Function)
+
+            Dim argumentsInEvaluationOrder As Lazy(Of ImmutableArray(Of IArgument)) = New Lazy(Of ImmutableArray(Of IArgument))(
+                Function()
+                    Return DeriveArguments(eventInvocation.Arguments, eventInvocation.Method.Parameters)
```

Comment:
>eventInvocation.Method [](start = 70, length = 22)

Should we expose symbol for the "fire" method in any way? For example, if there are no parameters, how one figures out what method is used?

Reply:
I think it can be retrieved from `IRaiseEventExpression.EventReference.Event.RaiseMethod`.

---
In reply to: [140929467](https://github.com/dotnet/roslyn/pull/22243#discussion_r140929467) [](ancestors = 140929467)

Reply:
> I think it can be retrieved from IRaiseEventExpression.EventReference.Event.RaiseMethod.

I believe there isn't one for a field-like event:
```
Event E As System.Action
```

---
In reply to: [142754335](https://github.com/dotnet/roslyn/pull/22243#discussion_r142754335) [](ancestors = 142754335,140929467)

Reply:
So you are talking about the delegate invoke methods? Can they be accessed through `eventSymbol.AssociateField`? 
Filed https://github.com/dotnet/roslyn/issues/22586